File: LeetCode/92. Reverse Linked List II/Solution.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function reverseBetween(head: ListNode | null, left: number, right: number): ListNode | null {
    if(right === left)  return head;
    let current : ListNode = new ListNode();
    current.next = head;
    head = current;

    for(let i : number = 1 ; i < left; i++) current = current.next;

    var prev = current;
    current = current.next;
    for(let i : number = 0; i < right - left; i++){
        let next : ListNode = current.next;
        
        current.next = next.next;
        next.next = prev.next;
        prev.next = next;
    }

    return head.next; 
};

File: LeetCode/92. Reverse Linked List II/Solution.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution
{
    public ListNode ReverseBetween(ListNode head, int left, int right)
    {
        if (right - left + 1 == 1) return head;
        var current = new ListNode();
        current.next = head;
        head = current;

        for (int i = 1; i < left; i++) current = current.next;

        var prev = current;
        current = current.next;
        for (int i = 0; i < right - left; i++)
        {
            var next = current.next;

            current.next = next.next;
            next.next = prev.next;
            prev.next = next;
        }

        return head.next;
    }
}